README.md
# NAME

Datahub::Factory::Arthub - modules for the L<VKC Arthub|>

# DESCRIPTION

This modules contains packages that the L<VKC Arthub|> uses to extract data from project-specific sources.

It contains the following packages:

- [Datahub::Factory::Importer::EIZ](https://metacpan.org/pod/Datahub::Factory::Importer::EIZ)

    An importer for the Erfgoed Inzicht organisation based on Adlib / OAI.

- [Datahub::Factory::Importer::KMSKA](https://metacpan.org/pod/Datahub::Factory::Importer::KMSKA)

    An importer for the Royal Musuem of Fine Arts Antwerp based on TMS.

- [Datahub::Factory::Importer::VKC](https://metacpan.org/pod/Datahub::Factory::Importer::VKC)

    An importer for the Flemish Art Collection based on Collective Access.

- [Datahub::Factory::Exporter::](https://metacpan.org/pod/Datahub::Factory::Exporter::)

    An exporter for the Arthub Flanders platform absed on LIDO and OAI.

\-head1 KMSKA

The KMSKA module is based on the TMS installation operated by 
the Royal Museum of Fine Arts Antwerp. Before using the module in 
a Datahub::Factory pipeline, you need to execute a SQL file against 
the MySQL (MariaDB) database  which contains a replication of 
the TMS MS SQL database.

Run this command:

$ mysql -u &lt;username> -p &lt;databasename> < Resources/TMS/schema.sql

# AUTHORS

Matthias Vandermaesen <matthias.vandermaesen@vlaamsekunstcollectie.be>
Pieter De Praetere <pieter@packed.be>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by PACKED, vzw, Vlaamse Kunstcollectie, vzw.

This is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License, Version 3, June 2007.

# SEE ALSO

[Datahub::Factory](https://metacpan.org/pod/Datahub::Factory)
[Catmandu](https://metacpan.org/pod/Catmandu)


Changes
Revision history for Datahub-Factory-VKC

{{$NEXT}}

1.02  2018-09-10 11:37:20 CEST

    - Fixed: merge_call on objTitles throws uninit value error.
    - Changed: added missing alter on ObjTitles.

1.01  2018-09-07 17:00:39 CEST

    - Changed: added objtitles table as sqlite.

1.00  2018-09-03 14:18:59 CEST

    - Stable release.

0.100  2018-09-03 14:10:29 CEST

    - Changed: Added missing indexes on the CITvgsrpObjTombstoneD_RO table.
    - Changed: Skip record if no LidoRecID is available.

0.99  2018-08-01 16:24:13 CEST

    - Changed: Yanked out SQL code and put in separate SQL file.

0.98  2018-06-20 16:33:07 CEST

    - Changed: updated KMSKA modules for TMS 2017.

0.97  2018-05-03 15:27:40 CEST

    - Fixed: Fixes #2 Multiple LidoRecID's.

0.96  2017-12-06 13:51:39 CET

    - Fixed: Added missing prereqs as reported by CPANTS

0.95  2017-12-05 10:01:27 CET

    - Changed: removed stale PID module code.
    - Changed: removed creating temporary SQLite databases from EIZ module.

0.94  2017-11-13 11:28:19 CET

    - Changed: reverted back to 0.92 changes because bugs.

0.93  2017-11-08 13:17:10 CET

    - Added: added generate_temp_tables and generate_vocabularies plugin options.
    - Changed: remove stale PID module.
    - Changed: added AAT & Creators SQLite generation to KMSKA module.

0.92  2017-09-06 10:59:08 CEST

    - Changed: updated documentation and licensing

0.91  2017-09-05 14:18:12 CEST

    - Changed: improved import headers and our $VERSION.
    - Changed: minimal required version of Datahub::Factory is 1.70.
    - Added: a dist.ini for dist / dzil milla
    - fixed: missing Catmandu::Adlib dependency
    - changed: moved TMS::Index to a dedicated KMSKA namespace
    - added: missing dependency DBD::mysql

0.9  2017-06-08 11:27:39 CEST
    - changed: conform to new Catmandu::Store::Datahub API
    - added: Arthub Exporter module
    - changed: disabled Rackspace and Webfile support. Only local file supported. PIDs should be part of the registration system.
    - renamed: MSK importer to EIZ (ErfgoedInzicht) importer
    - changed: break if no CSV file was found
    - added: missing dependencies
    - updated: CHANGES file
    - added: missing required modules

0.8  2017-05-12 12:58:42 UTC
    - Improved error handling

0.07  2017-05-11 08:01:17 UTC
    - Updated documentation
    - Added extra PID options

0.06  2017-05-04 07:48:16 UTC
    - Updated documentation for MSK

0.05  2017-05-04 06:48:17 UTC
    - Updated documentation

0.04  2017-05-04 06:16:37 UTC
    - Added re-usable PID module

0.03  2017-04-20 08:05:34 UTC
    - PIDS are now in TMS

0.02  2017-03-28 11:40:32 UTC
    - Initial release (merge of Datahub::Factory::KMSKA, Datahub::Factory::MSK, Datahub::Factory::PIDS and Datahub::Factory::VKC)


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v6.010.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.28"
  },
  "configure_requires" => {
    "Module::Build" => "0.28"
  },
  "dist_abstract" => "modules for the L<VKC Arthub|>",
  "dist_author" => [
    "Matthias Vandermaesen <matthias.vandermaesen\@vlaamsekunstcollectie.be>"
  ],
  "dist_name" => "Datahub-Factory-Arthub",
  "dist_version" => "0.100",
  "license" => "gpl",
  "module_name" => "Datahub::Factory::Arthub",
  "recursive_test_files" => 1,
  "requires" => {
    "Authen::NTLM" => 0,
    "Catmandu" => 0,
    "Catmandu::AAT" => 0,
    "Catmandu::Adlib" => 0,
    "Catmandu::Importer::DBI" => 0,
    "Catmandu::Importer::OAI" => 0,
    "Catmandu::Store::DBI" => 0,
    "Catmandu::VIAF" => 0,
    "Config::Simple" => 0,
    "DBD::SQLite" => 0,
    "DBD::mysql" => 0,
    "DBI" => 0,
    "Datahub::Factory" => "1.70",
    "LWP::Protocol::https" => 0,
    "LWP::UserAgent" => 0,
    "Lido::XML" => 0,
    "Moo" => 0,
    "Try::Tiny" => 0,
    "URI::Split" => 0,
    "WebService::Rackspace::CloudFiles" => 0,
    "namespace::clean" => 0,
    "perl" => "5.008005"
  },
  "test_requires" => {
    "Test::More" => "0.96"
  }
);


my %fallback_build_requires = (
  "Module::Build" => "0.28",
  "Test::More" => "0.96"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);


$build->create_build_script;
